perception/elevation_map_loader/src/elevation_map_loader_node.cpp

```diff
@@ -170,6 +208,63 @@ void ElevationMapLoaderNode::onVectorMap(
   }
 }
 
+void ElevationMapLoaderNode::receiveMap()
+{
+  pcl::PointCloud<pcl::PointXYZ> map_pcl;
+  sensor_msgs::msg::PointCloud2 pointcloud_map;
+  // create a loading request with mode = 1
+  auto request = std::make_shared<autoware_map_msgs::srv::GetDifferentialPointCloudMap::Request>();
+  // request all area
+  request->area.type = autoware_map_msgs::msg::AreaInfo::ALL_AREA;
+  std::vector<std::string> cached_ids{};
+  if (!pcd_loader_client_->service_is_ready()) {
+    RCLCPP_INFO(
+      this->get_logger(),
+      "Waiting for pcd map loader service. Check if the enable_differential_load in "
+      "pointcloud_map_loader is set `true`.");
+  }
+  bool is_all_received = false;
+  while (!is_all_received) {
+    // update cached_ids
+    request->cached_ids = cached_ids;
+    // send a request to map_loader
+    RCLCPP_INFO(this->get_logger(), "send a request to map_loader");
+    auto result{pcd_loader_client_->async_send_request(
+      request,
+      [](rclcpp::Client<autoware_map_msgs::srv::GetDifferentialPointCloudMap>::SharedFuture) {})};
+    std::future_status status = result.wait_for(std::chrono::seconds(0));
+    while (status != std::future_status::ready) {
+      RCLCPP_INFO(this->get_logger(), "waiting response");
+      if (!rclcpp::ok()) {
+        return;
+      }
+      status = result.wait_for(std::chrono::seconds(1));
+    }
+
+    RCLCPP_INFO(this->get_logger(), "concat maps");
+    if (result.get()->new_pointcloud_with_ids.empty()) {
+      RCLCPP_INFO(this->get_logger(), "finish receiving");
+      is_all_received = true;
+    } else {
+      // concat maps
+      for (const auto & new_pointcloud_with_id : result.get()->new_pointcloud_with_ids) {
+        if (pointcloud_map.width == 0) {
+          pointcloud_map = new_pointcloud_with_id.pointcloud;
+        } else {
+          pointcloud_map.width += new_pointcloud_with_id.pointcloud.width;
+          pointcloud_map.row_step += new_pointcloud_with_id.pointcloud.row_step;
+          pointcloud_map.data.insert(
+            pointcloud_map.data.end(), new_pointcloud_with_id.pointcloud.data.begin(),
+            new_pointcloud_with_id.pointcloud.data.end());
+        }
+        cached_ids.push_back(new_pointcloud_with_id.cell_id);
+      }
+    }
```

Comment:
How about this?
```suggestion
    if (result.get()->new_pointcloud_with_ids.empty()) {
      RCLCPP_INFO(this->get_logger(), "finish receiving");
      is_all_received = true;
      break;

    // concat maps
    for (const auto & new_pointcloud_with_id : result.get()->new_pointcloud_with_ids) {
      if (pointcloud_map.width == 0) {
        pointcloud_map = new_pointcloud_with_id.pointcloud;
      } else {
        pointcloud_map.width += new_pointcloud_with_id.pointcloud.width;
        pointcloud_map.row_step += new_pointcloud_with_id.pointcloud.row_step;
        pointcloud_map.data.insert(
          pointcloud_map.data.end(), new_pointcloud_with_id.pointcloud.data.begin(),
          new_pointcloud_with_id.pointcloud.data.end());
      }
      cached_ids.push_back(new_pointcloud_with_id.cell_id);
    }
```

Reply:
Thank you!
However, IMO, flag(`is_all_received`) is more suitable than `break` for readability.
What do you think about it?

Reply:
Thank you for your review!
I added function concatenating pointcloud maps in https://github.com/autowarefoundation/autoware.universe/pull/3162/commits/ebe59189c85bfa6e9daac52f7c7b24a22fcc65bc, and now nested complexity depth decreased.